rules/CodeQuality/Rector/ClassMethod/DateTimeToDateTimeInterfaceRector.php

```diff
@@ -88,8 +88,16 @@ public function getNodeTypes(): array
     public function refactor(Node $node): ?Node
     {
         $isModifiedNode = false;
+        $dateTimeObject = new ObjectType('DateTime');
+
         foreach ($node->getParams() as $param) {
-            if (! $this->isObjectType($param, new ObjectType('DateTime'))) {
+            if (! $this->isObjectType($param, $dateTimeObject)) {
+                continue;
+            }
+
+            // Do not refactor if node's type is a child class of \DateTime (can have wider API)
+            $paramType = $this->nodeTypeResolver->resolve($param);
+            if (!$paramType->isSuperTypeOf($dateTimeObject)->yes()) {
```

Comment:
I am not in front of PC, you probably can use :

https://github.com/rectorphp/rector-src/blob/91ac830d085968bcd5faef806aabb663e50e1ff8/packages/NodeTypeResolver/NodeTypeResolver.php#L305

Reply:
could you try something like this :

```php
$this->nodeTypeResolver->isSameObjectTypes($paramType, [$dateTimeObject])
```

Reply:
I don't think so. At this point I would need `ObjectType` from the node's param but if it's nullable resolver would return `UnionType` which does not match that type. But maybe I don't get your idea 😉

Reply:
correct, you possibly need more fixtures like in #704 for more use case to ensure no regression:

- nullable DateTime native
- skip not object type
- skip not typed
- skip union type
- etc

Reply:
I'll look at it tomorrow, now it's time for a sleep 😉 Thanks for your feedback!

Reply:
Ok, I've added few more tests (taken from your PR) and they're green too without additional changes 👍 I've skipped string-based fixture because I think it duplicates the one with `stdClass` (since `string` also is a different type).